packages/wdio-sauce-service/src/service.js

```diff
@@ -90,6 +90,23 @@ export default class SauceService {
     }
 
     afterTest (test, context, results) {
+        /**
+         * remove failure if test was retried and passed
+         * > Mocha only
+         */
+        if (test._retriedTest && results.passed) {
+            --this.failures
```

Comment:
I mean do we need to have this one if https://github.com/webdriverio/webdriverio/pull/5997/files#r506921171

Reply:
Yes, the order is as follows:

- test fails: `this.failures++` (failure count: 1)
- test retries and fails: we return without adding to failure count
- test retries and fails: we return without adding to failure count
- test finally passes: we remove failure count

which resolves in having a total failure count of 0